### ISSUE TITLE
App build failed: Cannot find module '.output/public/__i18n__/prerender/3c199285.js'

### ISSUE THREAD

**Issue by lukaszflorczak** [nuxt-modules/i18n#2080]
### Environment

```
------------------------------
- Operating System: Darwin
- Node Version:     v18.16.0
- Nuxt Version:     3.4.3
- Nitro Version:    2.4.0
- Package Manager:  pnpm@8.5.1
- Builder:          vite
- User Config:      modules, i18n
- Runtime Modules:  @nuxtjs/i18n@8.0.0-beta.12
- Build Modules:    -
------------------------------
```

### Reproduction

Stackblitz: https://stackblitz.com/github/lukaszflorczak/nuxt-i18n-playground/tree/issue/vue-i18n-config
Repository: https://github.com/lukaszflorczak/nuxt-i18n-playground/tree/issue/vue-i18n-config

1. `pnpm install`
2. `pnpm build`
3. `node .output/server/index.mjs`

### Describe the bug

If I use a configuration file for vueI18n, there is an error during app building:
<img width="851" alt="Zrzut ekranu 2023-05-16 o 11 25 17" src="https://github.com/nuxt-modules/i18n/assets/1818979/95899922-7380-4c56-940a-58415c4483e9">

And also during preview:
<img width="2011" alt="Zrzut ekranu 2023-05-16 o 11 34 36" src="https://github.com/nuxt-modules/i18n/assets/1818979/1eab6fd9-3ad9-4afa-a30c-4fbe842dcad7">

It doesn't matter if I use the default file name `i18n.config.ts` or set a custom path with `i18n.vueI18n` in `nuxt.config.ts`.

**Comment by 50l3r** [comment]
Same problem:

Version: `8.0.0-beta.12`

Problem only appear on linux. Windows works well.

Im trying to put `type:module` on my package.json but it doesnt work.

**nuxt.config.ts**
```
[
    '@nuxtjs/i18n',
    {
        strategy: 'prefix_except_default',
        defaultLocale: 'es',
        locales: [
            {
                code: 'en',
                iso: 'es-ES'
            },
            {
                code: 'it',
                iso: 'it-IT'
            },
            {
                code: 'es',
                iso: 'es-ES'
            }
        ],
        detectBrowserLanguage: {
            useCookie: true,
            cookieKey: 'i18n_redirected',
            redirectOn: 'root' // recommended
        },

        vueI18n: './src/data/i18n/i18n.config.ts'
    }
]
```

**i18n.config.ts**
```
import es from './langs/es.json'
import en from './langs/en.json'
import it from './langs/it.json'

export default defineI18nConfig(() => ({
    legacy: false,
    locale: 'es',
    messages: {
        es,
        en,
        it
    }
}))
```

i18n.options.39f40470.js:1   GET xxxxxxxxxxx/__i18n__/prerender/de34d178.js net::ERR_ABORTED 500

**Comment by benjaminrobinet** [comment]
Same here 🫣

**Comment by 50l3r** [comment]
i revert to `8.0.0-beta.11` and problem doesnt appear but im suffer other error (only windows): https://github.com/nuxt-modules/i18n/issues/2039

**Comment by joelod** [comment]
Also having this issue.

**Comment by xibman** [comment]
As a workaround you can use this into nuxt.config.ts:

Put correct path to you node modules folder

```ts
vite: {
    build: {
      commonjsOptions: {
        include: ['../../../node_modules/**'],
        transformMixedEsModules: true,
      },
    },
  },
```

**Comment by erikuit** [comment]
It's not super clear from the changelog, but since 8.0.0-beta.12 this was added to the docs under getting started/setup:
![packagejson](https://github.com/nuxt-modules/i18n/assets/9920252/4ed8647e-29b5-44ef-b0a8-0f6bf8314f37)
After changing this it worked again for me.

**Comment by joelod** [comment]
@erikuit thanks, that did the trick on this side.

**Comment by kazupon** [comment]
Sorry, poor our changelog.
from v8.0.0-beta.12, we need to setup `"type": "module"` to `package.json`.

**Comment by monkemedia** [comment]
Anyone having this issue when testing your error pages?

So I have added `"type": "module"` to `package.json` which fixes the error, however, when I purposely go to a missing page to test my 404 page, the issue appears again. Below is a log from vercel:

```@nuxtjs/i18n Cannot find module '/var/public/__i18n__/prerender/cfcb45fa.js' imported from /var/task/chunks/app/_nuxt/i18n.options-1476e7cb.mjs
Unhandled Promise Rejection 	{"errorType":"Runtime.UnhandledPromiseRejection","errorMessage":"Error: Page not found: /bhh","reason":{"message":"Page not found: /bhh","statusCode":404,"statusMessage":"Page not found: /bhh"},"promise":{},"stack":["Runtime.UnhandledPromiseRejection: Error: Page not found: /bhh","    at process.<anonymous> (file:///var/runtime/index.mjs:1189:17)","    at process.emit (node:events:525:35)","    at emit (node:internal/process/promises:149:20)","    at processPromiseRejections (node:internal/process/promises:283:27)","    at process.processTicksAndRejections (node:internal/process/task_queues:96:32)"]}
Unknown application error occurred
Runtime.Unknown```

**Comment by joelod** [comment]
@monkemedia not here

**Comment by 50l3r** [comment]
I have the same problem

`i18n.options.025c93dd.js:1 @nuxtjs/i18n Failed to fetch dynamically imported module: https://example.com/__i18n__/prerender/de34d178.js`

Version: `v8.0.0-beta.12`
Environment: `Docker version 23.0.1, build a5ee5b1`
S.O: `Rocky Linux 8.6`

my i18n config:

```
import es from './langs/es.json'
import en from './langs/en.json'
import it from './langs/it.json'

export default defineI18nConfig(() => ({
    legacy: false,
    locale: 'es',
    messages: {
        es,
        en,
        it
    }
}))
```

I put `"type": "module"` on my package.json but problem persist.

The error is intermittent and does not happen every time. I have also tried in incognito mode.

I develop locally with windows and the problem does not appear

**Comment by donguyen912** [comment]
> 

i have same issue. Are you resolve it? @50l3r

**Comment by 50l3r** [comment]
> > 
> 
> i have same issue. Are you resolve it? @50l3r

nope :(

**Comment by donguyen912** [comment]
> 

issue fixed with update @nuxtjs/i18n@8.0.0-beta.13, you try @50l3r

**Comment by 50l3r** [comment]
> > 
> 
> issue fixed with update @nuxtjs/i18n@8.0.0-beta.13, you try @50l3r

Much thanks @glorfindel9. beta.13 works fine
